File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/next/records/contact/PatchContactRecord.kt
package expo.modules.contacts.next.records.contact

import expo.modules.contacts.next.records.fields.DateRecord
import expo.modules.contacts.next.records.fields.EmailRecord
import expo.modules.contacts.next.records.fields.ExtraNameRecord
import expo.modules.contacts.next.records.fields.PhoneRecord
import expo.modules.contacts.next.records.fields.PostalAddressRecord
import expo.modules.contacts.next.records.fields.RelationRecord
import expo.modules.contacts.next.records.fields.UrlAddressRecord
import expo.modules.kotlin.apifeatures.EitherType
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record
import expo.modules.kotlin.types.Either
import expo.modules.kotlin.types.ValueOrUndefined

@OptIn(EitherType::class)
data class PatchContactRecord(
  @Field val givenName: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val middleName: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val familyName: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val prefix: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val suffix: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val phoneticGivenName: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val phoneticMiddleName: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val phoneticFamilyName: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val company: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val department: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val jobTitle: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val phoneticOrganizationName: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val note: ValueOrUndefined<String?> = ValueOrUndefined.Undefined(),
  @Field val emails: ValueOrUndefined<List<Either<EmailRecord.Patch, EmailRecord.New>>?> = ValueOrUndefined.Undefined(),
  @Field val phones: ValueOrUndefined<List<Either<PhoneRecord.Patch, PhoneRecord.New>>?> = ValueOrUndefined.Undefined(),
  @Field val dates: ValueOrUndefined<List<Either<DateRecord.Patch, DateRecord.New>>?> = ValueOrUndefined.Undefined(),
  @Field val postalAddresses: ValueOrUndefined<List<Either<PostalAddressRecord.Patch, PostalAddressRecord.New>>?> = ValueOrUndefined.Undefined(),
  @Field val relationships: ValueOrUndefined<List<Either<RelationRecord.Patch, RelationRecord.New>>?> = ValueOrUndefined.Undefined(),
  @Field val urlAddresses: ValueOrUndefined<List<Either<UrlAddressRecord.Patch, UrlAddressRecord.New>>?> = ValueOrUndefined.Undefined(),
  @Field val extraNames: ValueOrUndefined<List<Either<ExtraNameRecord.Patch, ExtraNameRecord.New>>?> = ValueOrUndefined.Undefined()
) : Record

File: packages/expo-contacts/ios/next/records/contact/PatchContactRecord.swift
import ExpoModulesCore

struct PatchContactRecord: Record {
  @Field var givenName: ValueOrUndefined<String?> = .undefined
  @Field var middleName: ValueOrUndefined<String?> = .undefined
  @Field var familyName: ValueOrUndefined<String?> = .undefined
  @Field var maidenName: ValueOrUndefined<String?> = .undefined
  @Field var nickname: ValueOrUndefined<String?> = .undefined
  @Field var prefix: ValueOrUndefined<String?> = .undefined
  @Field var suffix: ValueOrUndefined<String?> = .undefined
  @Field var phoneticGivenName: ValueOrUndefined<String?> = .undefined
  @Field var phoneticMiddleName: ValueOrUndefined<String?> = .undefined
  @Field var phoneticFamilyName: ValueOrUndefined<String?> = .undefined
  @Field var company: ValueOrUndefined<String?> = .undefined
  @Field var department: ValueOrUndefined<String?> = .undefined
  @Field var jobTitle: ValueOrUndefined<String?> = .undefined
  @Field var phoneticCompanyName: ValueOrUndefined<String?> = .undefined
  @Field var note: ValueOrUndefined<String?> = .undefined
  @Field var image: ValueOrUndefined<String?> = .undefined
  @Field var birthday: ValueOrUndefined<ContactDateNext?>  = .undefined
  @Field var nonGregorianBirthday: ValueOrUndefined<NonGregorianBirthday?>  = .undefined
  @Field var emails: ValueOrUndefined<[Either<PatchEmailRecord, NewEmailRecord>]?> = .undefined
  @Field var phones: ValueOrUndefined<[Either<PatchPhoneRecord, NewPhoneRecord>]?> = .undefined
  @Field var dates: ValueOrUndefined<[Either<PatchDateRecord, NewDateRecord>]?> = .undefined
  @Field var addresses: ValueOrUndefined<[Either<PatchPostalAddressRecord, NewPostalAddressRecord>]?> = .undefined
  @Field var relations: ValueOrUndefined<[Either<PatchRelationRecord, NewRelationRecord>]?> = .undefined
  @Field var urlAddresses: ValueOrUndefined<[Either<PatchUrlAddressRecord, NewUrlAddressRecord>]?> = .undefined
  @Field var imAddresses: ValueOrUndefined<[Either<PatchImAddressRecord, NewImAddressRecord>]?> = .undefined
  @Field var socialProfiles: ValueOrUndefined<[Either<PatchSocialProfileRecord, NewSocialProfileRecord>]?> = .undefined
}
